Sources/SwiftSyntaxBuilderGeneration/Node.swift.gyb

```diff
@@ -0,0 +1,125 @@
+}%
+//// Automatically Generated From Node.swift.gyb.
+//// Do Not Edit Directly!
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// A Syntax node, possibly with children.
+/// If the kind is "SyntaxCollection", then this node is considered a Syntax
+/// Collection that will expose itself as a typedef rather than a concrete
+/// subclass.
+public class Node {
+  public let syntaxKind: String
+  public let swiftSyntaxKind: String
+  public let name: String
+  public let description: String?
+  public let baseKind: String
+  public let traits: [String]?
+  public let children: [Child]?
+  public let collectionElementName: String?
+  public let collectionElementChoices: [String]?
+  public let omitWhenEmpty: Bool
+  public let elementsSeparatedByNewline: Bool
+  public let collectionElement: String
+
+  public var baseType: String {
+    if self.baseKind == "SyntaxCollection" {
+      return "Syntax"
+    } else {
+      return kindToType(kind: baseKind)
+    }
+  }
+
+
+  public var collectionElementType: String {
+    return kindToType(kind: collectionElement)
+  }
+
+  /// Returns `True` if this node declares one of the base syntax kinds.
+  public var isBase: Bool {
+    return SyntaxBaseKinds.contains(syntaxKind)
+  }
+
+  /// Returns `True` if this node is a subclass of SyntaxCollection.
+  public var isSyntaxCollection: Bool {
+    return self.baseKind == "SyntaxCollection"
+  }
+
+  /// Returns `True` if this node should have a `validate` method associated.
+  public var requiresValidation: Bool {
+    return isBuildable
+  }
+
+  /// Returns `True` if this node is an `Unknown` syntax subclass.
+  public var isUnknown: Bool {
+    return syntaxKind.contains("Unknown")
+  }
+
+  /// Returns `True` if this node should have a builder associated.
+  public var isBuildable: Bool {
+    return !isBase && !isUnknown && !isSyntaxCollection
+  }
+
+  /// Returns 'True' if this node shall not be created while parsing if it has no children.
+  public var shall_be_omitted_when_empty: Bool {
+    return omitWhenEmpty
+  }
+
+  /// Returns true if this child has a token kind.
+  public var isToken: Bool {
+    return syntaxKind.contains("Token") || collectionElement.contains("Token")
+  }
+
+  public init(name: String,
+              description: String? = nil,
+              kind: String,
+              traits: [String]? = nil,
+              children: [Child]? = nil,
+              element: String? = nil,
+              elementName: String? = nil,
```

Comment:
if `elementName` is `nil` it default to `element`. 
So if we make it non-option, it will have a different behaviour. 

https://github.com/apple/swift/blob/528bd11db8a215255a3b50eca36bd72a37551f68/utils/gyb_syntax_support/Node.py#L47

Reply:
You’re right. I thought you always pass in a string literal and missed that 
```swift
parameters += ['elementName: "%s"' % node.element_name]
```
is guarded by an `if`